client/protocols/analytics.py

```diff
@@ -21,6 +25,70 @@ def run(self, messages):
             # TODO(denero) Get the canonical name of the question
             statistics['question'] = self.args.question
 
+        statistics['started'] = self.check_start(messages['file_contents'])
+        print("DEBUG:" + statistics['started'])
+
         messages['analytics'] = statistics
 
+    def check_start(self, files):
+        """returns a dictionary where the key is question number, and the value
+        signals whether the question has been started.
+        """
+        begin = re.compile(r"""
+            [ \t]*
+            \#[ ]BEGIN[ ](.+)    # \1 is the question tag
+            """, re.X | re.M)
+        end = re.compile(r"""
+            [ \t]*
+            \#[ ]END[ ](.+)    # \1 is the question tag
+            """, re.X | re.M)
+        replace = re.compile(r"""
+            \#[ ]Replace[ ].+
+            """, re.X | re.M)
+
+        q_status = {}
+
+        for path in files:
+            lines = files[path].splitlines()
+            if len(lines) == 0:
+                log.warning("File {0} has no content".format(path))
+            line_num = 0
+            started = False
+            in_block = False
+            prev_begin_tag = None
+
```

Comment:
@albert12132 This method works for the case when there is not ALT codes between the two tags. I was trying to expand this into something that would cope with both situations:
```
\s*[\#\;]\s+BEGIN\s+(.*?)\n   # \1 is question name
(.*)                          # \2 is the contents in between
\s*[\#\;]\s+END\s+\1
```
I would use this to find all the code snippets between tags, and group \2 to check it it is the ALT code that hasn't been changed. But practically, this regex wouldn't work for the question missing the END tag; it would just match one tag, and \2 would include everything till the end of the file. Any suggestions on how to make the regex work?